swap/src/bitcoin/wallet.rs

```diff
@@ -7,55 +7,100 @@ use crate::{
     execution_params::ExecutionParams,
 };
 use ::bitcoin::{util::psbt::PartiallySignedTransaction, Txid};
-use anyhow::{Context, Result};
+use anyhow::{anyhow, bail, Result};
 use async_trait::async_trait;
 use backoff::{backoff::Constant as ConstantBackoff, tokio::retry};
-use bitcoin_harness::{bitcoind_rpc::PsbtBase64, BitcoindRpcApi};
-use reqwest::Url;
-use std::time::Duration;
-use tokio::time::interval;
+use bdk::{
+    blockchain::{noop_progress, Blockchain, ElectrumBlockchain},
+    electrum_client::{Client, ElectrumApi},
+    keys::GeneratableDefaultOptions,
+    FeeRate,
+};
+use reqwest::{Method, Url};
+use serde::{Deserialize, Serialize};
+use std::{path::Path, sync::Arc, time::Duration};
+use tokio::{sync::Mutex, time::interval};
+
+const SLED_TREE_NAME: &str = "default_tree";
 
-#[derive(Debug)]
 pub struct Wallet {
-    pub inner: bitcoin_harness::Wallet,
+    pub inner: Arc<Mutex<bdk::Wallet<ElectrumBlockchain, bdk::sled::Tree>>>,
```

Comment:
The intent was for Bob to still use a `bitcoind`. It seems we are using BDK for both Alice and Bob. Is this intentional?

Reply:
Clarification: We intend for Alice to use `bitcoind` - as she is the role within Nectar. 

Since Nectar was not introduced yet we opted for changing both parties and then potentially bring back `bitcoind` for Alice again. (we discussed that in the chat)
I am not sure this strategy is optimal, as it might block merging https://github.com/comit-network/xmr-btc-swap/pull/177

@thomaseizinger what do you think? Should we just bring back `bitcoind` and maintain both wallets (one for CLI = Bob, one for Nectar = Alice)?

Reply:
Like we discussed on Friday, I wouldn't bother with bitcoind. Makes deploying Alice easier.